python/doc/examples/numerical_methods/general_methods/plot_estimate_integral_Gauss_Kronrod.py

```diff
@@ -0,0 +1,192 @@
+
+# %%
+# Notice that the algorithm can integrate a function which has several outputs (but the number of inputs is restricted to 1).
+# This is why we use the index `[0]` of `computed`, since :meth:`~openturns.GaussKronrod.integrate` returns a :class:`~openturns.Point`.
+# In order to check this computation, we compute the log-relative error in base 10.
+# In most cases (except when the exponent of the two numbers are different), this represents the number of correct digits in base 10.
+expected = 1.154700538379251529
+LRE_10 = -np.log10(abs(computed[0] - expected) / abs(expected))
+LRE_10
+
+# %%
+# The method computes more than 14 digits correctly. Given that 17 digits is the best we can, this is an astonishing performance.
+#
+# We then compute the number of function evaluations.
+after_evaluation_number = integrand.getEvaluationCallsNumber()
+number_of_calls = after_evaluation_number - before_evaluation_number
+number_of_calls
+
+# %%
+# Advanced use
+# ============
+#
+# The Gauss-Kronrod algorithm strives to produce an approximated integral which actual error is less than the tolerance.
+# The algorithm estimates the error, which may be used to guess the accuracy in the situation where the exact value is unknown (this is the general use case, of course).
+#
+# In order to get the error estimated by the algorithm, we use the third input argument of the :meth:`~openturns.GaussKronrod.integrate` method.
+error = ot.Point()
+computed = algo.integrate(integrand, interval, error)
+computed[0]
+
+# %%
+# The variable `error` now contains the error estimate from the algorithm.
+error[0]
+
+# %%
+# We see that the error estimate is a little lower than the tolerance, which indicates that the integral should be correctly approximated.
+error = ot.Point()
+lowerBound = 0.0
+upperBound = 1.0
+ai = ot.Point()
+bi = ot.Point()
+ei = ot.Point()
+fi = ot.Sample()
+computed = algo.integrate(
+    integrand, lowerBound, upperBound, error, ai, bi, fi, ei)
+computed[0]
+
+# %%
+# The error still contains the estimate of the error.
+error[0]
+
+# %%
+# During the algorithm, a collection of subintegrals
+#
+# .. math::
+#    \int_{a_i}^{b_i} g(x) dx
+#
+# are approximated.
+# The outputs :math:`a_i` and :math:`b_i` contain the subintervals used in the algorithm.
+print('ai:', ai)
+print('bi:', bi)
+
+# %%
+# The corresponding value of the integrals are in :math:`f_i`.
+# Since :math:`f` can be a multidimensional point, this is a :class:`~openturns.Sample`, which dimension corresponds to the output dimension of the function :math:`f`.
+print('fi:', fi)
+
+# %%
+# The sum of these sub-integrals is the value of the integral:
+sum(fi)[0]
+
+# %%
+# The estimated error of each integral is in :math:`e_i`:
+print(ei)
+number_of_intervals = ai.getDimension()
+print('number of intervals:', number_of_intervals)
+for i in range(number_of_intervals):
+    print(
+        "Integral #%d : [%.4f, %.4f] is %.4f with error %.3e"
+        % (i, ai[i], bi[i], fi[i, 0], ei[i])
```

Comment:
```suggestion
        f"Integral #{i} : [{ai[i]:.4f}, %.4f] is %.4f with error %.3e"
        % (i, ai[i], bi[i], fi[i, 0], ei[i])
```
Not to be applied but completed.